### PR TITLE
Refs #33029 -- Fixed popups Selenium tests in headless mode.

### PR DESCRIPTION
**Pull request by carltongibson**
- Adjusted Selenium test for headless Firefox.
   Ensure select element is fully inside viewport.

- ...

### CODE REVIEW

**Comment by carltongibson** [comment]
buildbot, test on selenium.

**Comment by carltongibson** [comment]
Rebased after #14943

**Comment by carltongibson** [comment]
buildbot, test on selenium.

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -5046,11 +5052,13 @@ def test_related_popup_incorrect_close(self):
 
         self.selenium.switch_to.window(test2_window)
         self.selenium.find_element_by_xpath('//input[@value="Save"]').click()
+        self.wait_until(lambda d: len(d.window_handles) == 2, 1)
```
Comment (felixxm) [inline tests/admin_views/tests.py:5055]: Can we use [number_of_windows_to_be()](https://www.selenium.dev/selenium/docs/api/py/webdriver_support/selenium.webdriver.support.expected_conditions.html#selenium.webdriver.support.expected_conditions.number_of_windows_to_be)?
```suggestion
        self.wait_until(ec.number_of_windows_to_be(2))
```
Also, we can probably chop `self.assertEqual(len(self.selenium.window_handles), 2)` in the next line.

Comment (carltongibson) [reply tests/admin_views/tests.py:5055]: > Also, we can probably chop self.assertEqual(len(self.selenium.window_handles), 2) in the next line.

The `wait_until` can timeout without the condition being true, so the assertion catches the closing not working. (That's not likely, but it's what the test case is about...)

**Comment by carltongibson** [comment]
buildbot, test on selenium.

**Pull request merged**